router/internal/fabric/manager.go

```diff
@@ -83,32 +89,44 @@ func GetStateManager() StateManager {
 	}
 
 	singleStateManager = &StateManagerImpl{
-		EventEmmiter:           events.New(),
-		networkSessionsByToken: &sync.Map{},
-		sessionsByToken:        &sync.Map{},
-		supportedFabrics:       &sync.Map{},
-		activeSessions:         &sync.Map{},
+		EventEmmiter:       events.New(),
+		sessionsByToken:    &sync.Map{},
+		apiSessionsByToken: &sync.Map{},
+		supportedFabrics:   &sync.Map{},
+		activeApiSessions:  &sync.Map{},
 	}
 
 	return singleStateManager
 }
 
 func (sm *StateManagerImpl) AddApiSession(session *edge_ctrl_pb.ApiSession) {
 	pfxlog.Logger().Debugf("adding session [%s] fingerprints [%s]", session.Token, session.CertFingerprints)
-	sm.sessionsByToken.Store(session.Token, session)
+	sm.apiSessionsByToken.Store(session.Token, session)
 	sm.Emit(EventAddedSession, session)
 }
 
 func (sm *StateManagerImpl) UpdateApiSession(session *edge_ctrl_pb.ApiSession) {
 	pfxlog.Logger().Debugf("updating session [%s] fingerprints [%s]", session.Token, session.CertFingerprints)
-	sm.sessionsByToken.Store(session.Token, session)
+	sm.apiSessionsByToken.Store(session.Token, session)
+
```

Comment:
I don't know the frequency of this is, but would it make sense to have a second id -> token map, so that we could do this with two map lookups, rather than having to scan the map?